tests/src/python/test_provider_ogr_general.py

```diff
@@ -0,0 +1,980 @@
+                        self.assertEqual(polygon_area, polygon_area_check)
+                        print('-I-> Polygon count[{0}] polygon_area[{1:f}] = (exterior_ring_area[{2:f}] minus internal_ring_area_sum[{3:f}])'.format(index, polygon_area,exterior_ring_area,internal_ring_area_sum))
+                    else:
+                        print('-I-> Polygon[{0}] area[{1:f}]: ExternalRing-Area[{2:f}] ; InteriorRings[{3}]'.format(index, polygon_area, exterior_ring_area, count_rings))
+
+                geom_area_sum_check=exterior_ring_area_sum-internal_ring_area_sum_total
+                geom_area_sum_diff=geom_area_sum-geom_area_sum_check;
+                print('-I-> diff [{0:f}]  '.format(geom_area_sum_diff))
+                print('-I-> MultiPolygon geom_area_sum_check[{0:f}] = exterior_ring_area_sum_total[{1:f}] minus internal_ring_area_sum_total[{2:f}])'.format(geom_area_sum_check, exterior_ring_area_sum, internal_ring_area_sum_total))
+                print('-I-> MultiPolygon count[{0}] MultiPolygon-Area[{1:f}] Polygon-Areas[{2:f}] Ring-Areas[{3:f}] '.format(count_polygons,geom_area,geom_area_sum,geom_area_sum_check))
+                self.assertEqual(geom_area,geom_area_sum)
+                self.assertEqual(round(geom_area_sum,5),round(geom_area_sum_check,5))
+                # -I-> MultiPolygon count[13] MultiPolygon-Area[87622761.119470] Polygon-Areas[87622761.119470] Ring-Areas[87622761.119469]
+                # SELECT ROUND(ST_Area(soldner_polygon),5) AS area_polygon, ST_Area(soldner_ring) AS area_polygon,* FROM pg_bezirke_1938 WHERE id_admin=1902010800;
+                # 87622761.119469	87622761.119469
+                print('-I-> MultiPolygon-Area check correct [does not include the area of the Internal-Rings[{0:f}])'.format(internal_ring_area_sum_total))
+
+        print('-I-> test_03_{0} [{1}]'.format(test_name,'end of test'))
+
+###############################################################################
+# Test GML with more than 1 geometry
+# - 2 files are being checked
+# - 1 with 2 POINTs with 1 record
+# --  check if given values of POINT are correct
+# - 1 created from a spatialite TABLE with
+# --> ogr2ogr -f GML gdal_220.ogr_Rauletshof_1909.gml admin_geometries_11.db -sql "SELECT * FROM admin_geometries"
+# - TABLE contains imformation about 6 historical streets, with metadata
+# -- the outline of the street as a POLYGON
+# -- the center of the street as a LINESTRING
+# -- the center of the street LINESTRING as a POINT
+# - Goals are:
+# --  read record 2 (Raulet's Hof) for the 3 types
+# --- determin correct Geometry-Type, testing that the correct Geometry-Field index is being used
+# ---  check value of POINT (or first POINT of LINESTRING or of ExternalRing of POLYGON)
+# Note: date fields are being returned as strings
+
+    def test_04_OgrGMLMultipleGeometries(self):
```

Comment:
What does this test (regarding QGIS code) that test_01_OgrSpatialTableMultipleGeometries wouldn't test ? I'd have a preference for the GML based tests since GML is text editable